packages/kvib-react-button/README.md

```diff
@@ -0,0 +1,32 @@
+# Button (React)
```

Comment:
Dette gjelder hele kvib-react-button:

Jeg er usikker på om vi bør gå for individuelle pakker for react-komponenter, versus det å ha alt samlet i packages/react. Det er en evig diskusjon som det finnes uendelige pros and cons på der ute, men her er nøkkelmomentene jeg tenker er relevant for oss:

- Fordelen med flere pakker er at utviklere kan selv velge hvilke komponenter de vil oppdatere når
  - ...men nedsiden her er at CSS'en er ikke modulær, så vi vil ørten forskjellige versjoner av hele CSS-biblioteket, det er fare for at komponenter blir inkompatible med hverandre (f.eks. hvis vi Icon som en pakke og Button som en annen, og Button skal kunne inkludere Icon, så kan man få peer-dependency-helvete)
- Ironisk nok kan da flere komponenter føre til større pakkestørrelse enn når man har ett bibliotek. Når alt er samlet i @kvib/react så får man fordelene med tree shaking og alt det der på ting du ikke bruker, og man er i hovedsak avhengig av CSS-pakken.
  - Når man har flere react-pakker så får man flere CSS-pakker som må inkluderes, mye ubrukt kode
  - (jeg tror ikke vi trenger å bry oss så mye om pakkestørrelse for å være ærlig, så dette argumentet er litt samme det)
- Den verste biten er overhead. Flere pakker å holde styr på, praksis på de forskjellige pakkene vil divergere etterhvert som forskjellige folk bidrar på forskjellige ting

**TL;DR:** Anbefaler å ta utgangspunkt i kvib/css og holde alt i én pakke fremfor å dele opp i flere. Mindre overhead, mindre kompleksitet, mindre friksjon for at vi skal få i gang dette designsystemet.

Reply:
Jeg har personlig ingen sterke meninger om dette. Jeg har brukt bibliotek av begge typer. Dog aldri bidratt til å utvikle og arbeide med noen. Vi gikk bare for dette fordi Spor har gått for denne løsningen.

Men du har mange gode poeng og hvis ingen har noen motforestillinger så er det vel ingenting i veien å bare kjøre på i den retningen tenker jeg.

Eneste ulempene jeg ser med å ha én pakke er selvsagt antallet releaser for hver eneste lille patch endring vi gjør... Men det er kanskje ikke så farlig. Jeg vet ikke.